src/item.cpp

```diff
@@ -5983,55 +5983,55 @@ bool item::allow_crafting_component() const
 }
 
-int item::get_static_temp_counter() const
-{
-    int counters = item_counter;
-    if( item_tags.count( "FROZEN" ) ) {
-        counters = -1200 - counters;
-    } else if( item_tags.count( "COLD" ) ) {
-        counters = -600 - counters;
-    } else if( item_tags.count( "HOT" ) ) {
-        counters += 600;
-    } else if( !item_tags.count( "WARM" ) ) { // if not warm, then ticking for cold
-        counters = -counters;
+
+float item::get_temp_from_energy( const float true_energy )
+{    
+    const int specific_heat_liquid = type->comestible->spec_heat_liquid; // J/g K
+    const int specific_heat_solid = type->comestible->spec_heat_solid; // J/g K
+    const int latent_heat = type->comestible->lat_heat; // J/kg
+    const float mass = to_gram( weight() ); // g
+    const float freezing_temperature = ( type->comestible->freeze_point - 32 ) * 0.556 + 273.15;  // F converted to K
+    const float completely_frozen_energy = mass * specific_heat_solid * freezing_temperature;  // Energy that the item would have if it was completely solid at freezing temperature
+    const float completely_liquid_energy = completely_frozen_energy + mass * latent_heat; // Energy that the item would have if it was completely liquid at freezing temperature
+    float new_item_temperature;
+    
+    if ( true_energy > completely_liquid_energy ) {
+        // Item is liquid
+        new_item_temperature = freezing_temperature + ( true_energy - completely_liquid_energy ) / ( specific_heat_liquid * mass );
+    } else if ( true_energy < completely_frozen_energy ) {
+        // Item is solid
+        new_item_temperature = true_energy / ( specific_heat_solid * mass );
+    } else {
+        // Item is partially solid
+        new_item_temperature = freezing_temperature;
     }
-    return counters;
+    return new_item_temperature;
 }
 
-void item::set_temp_from_static( const int counters )
-{
-    item_tags.erase( "COLD" );
-    item_tags.erase( "HOT" );
-
-    if( counters < -1200 ) {
-        item_tags.insert( "FROZEN" );
-        item_counter = -counters - 1200;
-        if( current_phase == LIQUID ) {
-            current_phase = SOLID;
+float item::get_energy_from_temperature( const float new_temperature )
+{    
+    const int specific_heat_liquid = type->comestible->spec_heat_liquid; // J/g K
+    const int specific_heat_solid = type->comestible->spec_heat_solid; // J/g K
+    const int latent_heat = type->comestible->lat_heat; // J/kg
+    const float mass = to_gram( weight() ); // g
+    const float freezing_temperature = ( type->comestible->freeze_point - 32 ) * 0.556 + 273.15;  // F converted to K
+    const float completely_frozen_energy = mass * specific_heat_solid * freezing_temperature;  // Energy that the item would have if it was completely solid at freezing temperature
+    const float completely_liquid_energy = completely_frozen_energy + mass * latent_heat; // Energy that the item would have if it was completely liquid at freezing temperature
```

Comment:
You calculate freezing temperature and energies twice.

Reply:
The already existing temp_to_kelvin function uses and returns double. I'll just make another fahrenheit_to_kelvin function that uses floats and use that for the temperature unit conversions.

The completely_[state]_energy is now calculated only twice (and completely_[state]_specific_energy once). Not worth it to make function for that.